test/snippet/CMakeLists.txt

```diff
@@ -37,14 +37,41 @@ macro (seqan3_snippet target_cpp snippet_full_path)
     unset (target)
 endmacro ()
 
+macro (seqan3_doc_snippet target_cpp snippet_full_path)
+    string (REGEX REPLACE ".cpp$" "" target_name ${target_cpp})
+    string (REGEX REPLACE ".+/doc/" "" source_path ${snippet_full_path})
```

Comment:
```suggestion
    string (REGEX REPLACE ".+/(doc|test)/" "" test_path ${snippet_full_path})
    string (REGEX REPLACE ".+/(doc|test/snippet)/" "" source_path ${snippet_full_path})
```

Reply:
I tried something similar but the problem is that when adding the executable you use the relative path $(source_path) but this does not work for doc snippets instead I needed the $(snippet_full_path) to make it work... not sure how to adjust this in a generic way as I am inexperienced with cmake :/